src/cace/util/threadset.c

```diff
@@ -56,6 +56,12 @@ int cace_threadset_join(cace_threadset_t tset)
     {
         pthread_t thr;
         cace_threadset_pop_back(&thr, tset);
+        if (pthread_cancel(thr))
```

Comment:
This will not allow the work thread to clean itself up properly and should not be used. The specific thread function should have a way to properly, cleanly return when the daemon is stopped.